src/main/java/uk/ac/ebi/embl/converter/cli/Main.java

```diff
@@ -10,158 +10,97 @@
  */
 package uk.ac.ebi.embl.converter.cli;
 
-import ch.qos.logback.classic.Level;
-import ch.qos.logback.classic.LoggerContext;
-import io.vavr.Function0;
-import java.io.*;
-import java.nio.charset.Charset;
-import java.nio.charset.StandardCharsets;
-import java.nio.file.Files;
-import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import picocli.CommandLine;
 import picocli.CommandLine.*;
-import picocli.CommandLine.Model.CommandSpec;
-import uk.ac.ebi.embl.converter.ConversionError;
-import uk.ac.ebi.embl.converter.Converter;
-import uk.ac.ebi.embl.converter.fftogff3.FFToGff3Converter;
-import uk.ac.ebi.embl.converter.gff3toff.Gff3ToFFConverter;
+import uk.ac.ebi.embl.converter.exception.CLIException;
+import uk.ac.ebi.embl.converter.exception.ExitException;
+import uk.ac.ebi.embl.converter.validation.RuleSeverity;
+import uk.ac.ebi.embl.converter.validation.ValidationRule;
 
 @Command(
         name = "gff3tools",
-        subcommands = {CommandConversion.class, CommandLine.HelpCommand.class},
+        subcommands = {FileConversionCommand.class, CommandLine.HelpCommand.class},
         description = "Utility to convert and validate gff3 files")
 public class Main {
-    @Spec
-    CommandSpec spec;
-
     private static final Logger LOG = LoggerFactory.getLogger(Main.class);
 
     public static void main(String[] args) {
-        int exitCode = -1;
+        int exitCode = 1;
         try {
-            exitCode = new CommandLine(new Main()).execute(args);
-        } catch (Exception e) {
+            exitCode = new CommandLine(new Main())
+                    .setExecutionExceptionHandler(new ExecutionExceptionHandler())
+                    .execute(args);
+        } catch (OutOfMemoryError e) {
+            String filename = new java.io.File(Main.class
+                            .getProtectionDomain()
+                            .getCodeSource()
+                            .getLocation()
+                            .getPath())
+                    .getName();
+            LOG.error(
+                    "The conversion needs more memory please increase the memory using the -Xmx java argument.\neg. java -jar -Xmx2G %s %s"
+                            .formatted(filename, Arrays.stream(args).collect(Collectors.joining(" "))));
```

Comment:
Can't this be moved to handleExecutionException method? Anyway, we must update the following
`java -jar -Xmx2G` to `java -Xmx2G -jar `